### ISSUE TITLE
Constant arrays perform differently in mrustc and rustc

### ISSUE THREAD

**Issue by danielrh** [thepowersgang/mrustc#106]
in my inner loop function I was referencing
https://github.com/dropbox/rust-brotli-decompressor/blob/master/src/prefix.rs#L124
```rust
pub const kCmdLut: [CmdLutElement; 704] = [CmdLutElement {
                                             insert_len_extra_bits: 0x00,
                                             copy_len_extra_bits: 0x00,
                                             distance_code: 0,
                                             context: 0x00,
                                             insert_len_offset: 0x0000,
                                             copy_len_offset: 0x0002,
                                           },
.....
```
called from https://github.com/dropbox/rust-brotli-decompressor/blob/45e53e3747bcac834af9351088ad418780d46670/src/decode.rs#L2047

which worked fast using rustc
but mrustc decided to inline the 704 element struct array into each function call, and this caused a 5x slowdown in the program.
The workaround to fix performance was changing `const` to `static` but rustc must have some sort of logic to avoid inlining the constants for arrays.
Even changing smaller arrays like
https://github.com/dropbox/rust-brotli-decompressor/blob/45e53e3747bcac834af9351088ad418780d46670/src/decode.rs#L55
to be static arrays had a positive benefit

My perf fixes for `mrustc` are committed to a branch https://github.com/dropbox/rust-brotli-decompressor/tree/simplified_mrustc_fixes

**Comment by thepowersgang** [comment]
As @bjorn3 implied, this is because mrustc doesn't do lifting and de-duplication of constants into statics.

**Comment by thepowersgang** [comment]
Should now be fixed (as of 6a86b6db7458d7310880b7867c5a2cce80bc8802). mrustc now convers shared borrows of constant values into statics. There isn't any de-duplication of these yet, not sure if that will be needed.
